### PR TITLE
Remove use_simd option from the public API

### PR DESCRIPTION
**Pull request by shssoichiro**
This should be a development-specific option.
This adds an environment variable, `AV_METRICS_DISABLE_SIMD`,
which can be set (to any value) to disable SIMD.

This is a breaking change.

### CODE REVIEW

File: av_metrics/benches/bench.rs [hunk 1]
```diff
@@ -62,19 +63,21 @@ pub fn msssim_benchmark(c: &mut Criterion) {
 pub fn ciede2000_nosimd_benchmark(c: &mut Criterion) {
     let frame1 = get_video_frame::<u8>("./testfiles/yuv420p8_input.y4m");
     let frame2 = get_video_frame::<u8>("./testfiles/yuv420p8_output.y4m");
+    env::set_var("AV_METRICS_DISABLE_SIMD", "1");
```
Comment (lu-zero) [inline av_metrics/benches/bench.rs]: I really dislike this.

Comment (shssoichiro) [reply av_metrics/benches/bench.rs]: I'm not a fan either. I think the workaround would be to make a separate module to expose the private structs when benchmarks are enabled, like rav1e does. I'm not a fan of that approach either, but it is better than this. Can you think of any better options?

Comment (lu-zero) [reply av_metrics/benches/bench.rs]: I'd have a feature to expose to criterion the internals and set them as needed. It is a workaround to the fact benchmarks do not carry a specific feature (and we should probably check the issue about it on cargo).

**Comment by shssoichiro** [comment]
Due to the limited surface area of the public API, I found it simpler to make functions (enabled by `cfg[test]` or `feature = "bench"`) which replicate the normal behavior, but with simd disabled. This also required me to move the integration tests to be unit tests (which maybe they should have been anyway), because integration tests apparently don't enable `cfg[test]`.

**Pull request merged**